### ISSUE TITLE
SynonymAnalyzer.php in module-search is evaluating user input as a regex

### ISSUE THREAD

**Issue by dan-ding** [magento/magento2#25886]
### Preconditions (*)

2.3.2-xyz
2.3-develop
prolly others

### Steps to reproduce (*)

1. install magento
2. enable elasticsearch
3. do a weird search for something like: http://www.a
GET /catalogsearch/result/?q=http://www.a

### Expected result (*)

1. the search results screen saying the term couldn't be found

### Actual result (*)

1. error message
2. in production mode, error report is generated

The error that occurs is on line 96 of search/Model/SynonymAnalyzer.php:  
https://github.com/magento/magento2/blob/f6405d571110d736fefb7f27925c8748da83c7fa/app/code/Magento/Search/Model/SynonymAnalyzer.php#L96

the error message:
report.ERROR: Warning: preg_match(): Unknown modifier '/' in /var/www/html/vendor/magento/module-search/Model/SynonymAnalyzer.php on line 96

preg_match is evaluating the user input as a regex

**Comment by dan-ding** [comment]
@magento this can't be shown on an instance as it requires elasticsearch

**Comment by engcom-Charlie** [comment]
Hello @dan-ding 
We are not able to reproduce this issue on the latest 2.3-develop branch by provided steps.

**Manual testing scenario:**

1. istall magento
2. enable elasticsearch
3. do a weird search for something like: `http://www.a`
4. `GET /catalogsearch/result/?q=http://www.a`

**Result:**
The search results screen saying the term couldn't be found
**Browser:**
![image](https://user-images.githubusercontent.com/51680850/70151416-45d91380-16b4-11ea-917c-afee9ba4ee06.png)
**Postman:**
![image](https://user-images.githubusercontent.com/51680850/70151256-feeb1e00-16b3-11ea-9829-3be10a30001b.png)

So i have to close this issue.
Thank you for contribution and collaboration!

**Comment by wojtekn** [comment]
@magento-engcom-team this was closed because it wasn't reproduced. However, incorrect steps were used so this should be reopened. 

The problem can be reproduced only when synonym exists for given phrase. So to reproduce:

1. Ensure the synonym exists in SEO & Search -> Search Synonyms:
foo,bar"foobar",
2. With ES enabled as search backend, search for "Foo/Bar" in frontend.

Expected behavior would be search results page, either empty or one showing results.

Current behavior is the exception:

```
1 exception(s):
Exception #0 (Exception): Warning: preg_match(): Unknown modifier 'B' in /var/www/html/vendor/magento/module-search/Model/SynonymAnalyzer.php on line 96
```

**Comment by magento-engcom-team** [comment]
Hi @dan-ding. Thank you for your report. 
The issue has been fixed in magento/magento2#30318 by @wojtekn in 2.4-develop branch
Related commit(s):
 - [ce1263da089d5b8ca28bc8689480092db0924d33](https://github.com/magento/magento2/commit/ce1263da089d5b8ca28bc8689480092db0924d33)
 - [ccec05add626879219b858485d8c06aeecc1a145](https://github.com/magento/magento2/commit/ccec05add626879219b858485d8c06aeecc1a145)
 - [970d1793bc408ad99cf959197c77dec424f4cefc](https://github.com/magento/magento2/commit/970d1793bc408ad99cf959197c77dec424f4cefc)
 - [38c4a3e01ebc3dd0d18286efba20d03373deb830](https://github.com/magento/magento2/commit/38c4a3e01ebc3dd0d18286efba20d03373deb830)
 - [88f6101fe15eda61471822169670b744914fdf40](https://github.com/magento/magento2/commit/88f6101fe15eda61471822169670b744914fdf40)

The fix will be available with the upcoming 2.4.2 release.
